src/lib/ast/jsonHandler.ts

```diff
@@ -0,0 +1,57 @@
+import { Skill } from "../types";
+import { And } from "./and";
+import { SkillExpression } from "./skillExpression";
+import { Or } from "./or";
+import { Variable } from "./variable";
+import { SkillsRelations } from "./skillsRelation";
+
+// An internal type used within this file to handle the parse Json to SkillExpression
+type JsonExpression = {
+	operator: string;
+	skills: string[];
+};
+
+// Convert a SkillExpression to a Json format
+export function createJson(operator: string, terms: SkillExpression[]): string {			
+    let nestedVariable: string[] = [];

```

Comment:
Function seems to work as expected (recursively), but the naming of this variable seems not to fit as in general this does not store `nestedVariables`.
I would suggest to name it like `translatedTerms`, `jsonTerms`, `jsonifiedTerms`, ...